test/generator/string_param_generator.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Halide.h"
+
+using namespace Halide;
+
+namespace {
+class StringParam : public Halide::Generator<StringParam> {
+public:
+    GeneratorParam<std::string> param{"param", "add_one"};
+
+    Input<Buffer<float>> input{ "input", 1 };
+    Output<Buffer<float>> output{ "output", 1 };
+
+    void generate() {
+        Var x;
+        if (param.value() == "add_one") {
```

Comment:
I realize you want a simple test, but this is a very bad example that I fear people will mimic in new code; in this case, using a GeneratorParam<bool> or GeneratorParam<EnumType>, etc would be vastly more appropriate. Can we come up with a test example that actually *requires* the string type, and none of the other existing types will reasonably substitute?

Reply:
How about an RPN description of the computation to perform e.g: "3 x + y +". That just requires Halide::Internal::split_string to break it into tokens and a stack (`vector<Expr>`) to parse into some Expr to use as the RHS of a Func.

Reply:
SGTM

Reply:
I agree with @steven-johnson its a bad example, I added it just want to make sure buildbots are happy with this change. @abadams example is perfect will update the test accordingly.